codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -164,4 +183,51 @@ class FileCodesignVisitor {
       processManager: processManager,
     );
   }
+
+  /// Visit and codesign a binary with / without entilement.
+  ///
+  /// At this stage, the virtual [entitlementCurrentPath] accumulated through the recursive visit, is compared
+  /// against the paths extracted from [fileWithEntitlements], to help determine if this file should be signed
+  /// with entitlements.
+  Future<void> visitBinaryFile({required FileSystemEntity binaryEntity, required String entitlementParentPath}) async {
+    if (await isSymlink(binaryEntity.absolute.path, processManager)) {
+      return;
+    }
+
+    final String currentFileName = binaryEntity.absolute.path.split('/').last;
+    final String entitlementCurrentPath = '$entitlementParentPath/$currentFileName';
+
+    if (!fileWithEntitlements.contains(entitlementCurrentPath) &&
+        !fileWithoutEntitlements.contains(entitlementCurrentPath)) {
+      log.severe('The system has detected a binary file at $entitlementCurrentPath\n'
+          'but it is not in the entitlements configuartion files your provided \n'
+          'if this is a new engine artifact, please add it to one of the entilements.txt files\n');
+      throw CodesignException(fixItInstructions);
+    }
+    log.info('\n signing file at path ${binaryEntity.absolute.path}');
+    log.info('\n the virtual entitlement path associated with file is $entitlementCurrentPath');
+    log.info('\n the decision to sign with entitlement is ${fileWithEntitlements.contains(entitlementCurrentPath)}');
```

Comment:
is this meant to be in production?

Reply:
nvm, you're probably right, it's good to have this in the logs.

Reply:
actually, even better, why don't you move this log until after the next line, and log the value of `args`.

Reply:
umm I thought this would be helpful in tests to check whether we code signed each binary with entitlements or not. Should we find another way to obtain this information in tests?

Reply:
Oh sorry i didn't refresh the page, there are more comments from Christopher. Sure I kept the logs and moved the \n out of the way.